cmake/TaichiCore.cmake

```diff
@@ -287,7 +289,9 @@ endif()
 if (TI_WITH_OPENGL)
     add_subdirectory(external/SPIRV-Cross)
     target_include_directories(${CORE_LIBRARY_NAME} PRIVATE external/SPIRV-Cross)
-    target_link_libraries(${CORE_LIBRARY_NAME} spirv-cross-glsl spirv-cross-core)
+    target_link_libraries(${CORE_LIBRARY_NAME} spirv-cross-glsl)
+    target_link_libraries(${CORE_LIBRARY_NAME} spirv-cross-hlsl)
```

Comment:
Should add an separate TI_WITH_DX11 flag somewhere and use that to link hlsl cross compiler. Then we can add the cross compiler core if either SPIR-V based OpenGL or DX is used.

Reply:
Added a check for TI_WITH_DX11 and TI_WITH_OPENGL and enabling spirv_hlsl and/or spirv_cross as neded.